pgjdbc/src/main/java/org/postgresql/largeobject/BlobInputStream.java

```diff
@@ -335,6 +351,59 @@ public boolean markSupported() {
     return true;
   }
 
+  /**
+   * Skips over and discards {@code n} bytes of data from this input stream.
+   *
+   * <p>Due to the "sparse" implementation of Large Objects, this class allows skipping
+   * past the "end" of the stream. Subsequent reads will continue to return {@code -1}.
+   *
+   * @param n the number of bytes to be skipped.
+   * @return the actual number of bytes skipped which might be zero.
+   * @throws IOException  if an underlying driver error occurs.
+   *     In particular, this will throw if attempting to skip beyond the maximum length
+   *     of a large object, which by default is 4,398,046,509,056 bytes.
+   * @see java.io.InputStream#skip(long)
+   */
+  @Override
+  public long skip(long n) throws IOException {
+    if (n <= 0) {
+      // The spec does allow for skipping backwards, but let's not.
+      return 0;
+    }
+
+    try (ResourceLock ignore = lock.obtain()) {
+      LargeObject lo = getLo();
+      long loId = lo.getLongOID();
+
+      long targetPosition = absolutePosition + n;
+      if (targetPosition > limit || targetPosition < 0) {
+        targetPosition = limit;
+      }
+      long currentPosition = absolutePosition;
+      long skipped = targetPosition - currentPosition;
+
+      if (buffer != null && buffer.length - bufferPosition > skipped) {
+        bufferPosition += (int) skipped;
+      } else {
+        try {
+          if (targetPosition <= Integer.MAX_VALUE) {
+            lo.seek((int) targetPosition, LargeObject.SEEK_SET);
+          } else {
+            lo.seek64(targetPosition, LargeObject.SEEK_SET);
+          }
+        } catch (SQLException e) {
+          throw new IOException(
+              GT.tr("Can not skip stream for large object {0} by {1} (currently @{2})",
+                  loId, n, currentPosition),
+              e);
+        }
+        buffer = null;
```

Comment:
```suggestion
        buffer = null;
        try {
          if (targetPosition <= Integer.MAX_VALUE) {
            lo.seek((int) targetPosition, LargeObject.SEEK_SET);
          } else {
            lo.seek64(targetPosition, LargeObject.SEEK_SET);
          }
        } catch (SQLException e) {
          throw new IOException(
              GT.tr("Can not skip stream for large object {0} by {1} (currently @{2})",
                  loId, n, currentPosition),
              e);
        }
```

The `buffer` should be invalidated at the very beginning to avoid stale data in case of failure.

Reply:
If any of these methods throw then you don't need to guarentee what continuing to use the stream will do, but sure.